lib/redis_client.rb

```diff
@@ -692,6 +692,22 @@ def connect
       raise
     end
   end
+
+  def with_middlewares(method, *args, &block)
+    Middlewares.send(method, *args) do
+      call_local_middlewares(config.middlewares, 0, method, *args, &block)
+    end
+  end
+
+  def call_local_middlewares(middlewares, position, method, *args, &block)
+    middleware = middlewares[position]
+
+    return block.call if middleware.nil?
+
+    middleware.send(method, *args) do
+      call_local_middlewares(middlewares, position + 1, method, *args, &block)
+    end
```

Comment:
This kind of recursion is precisely the type of performance traps I wanted to avoid. With argument dynamic splatting at each call, it often end up stupidly slow.